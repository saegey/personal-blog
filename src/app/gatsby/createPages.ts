import type { GatsbyNode } from 'gatsby'
import path from 'path'

type MdxData = {
  data?: {
    allMdx: {
      nodes: {
        id: string
        fields: {
          slug: string
        }
        frontmatter: {
          related: Array<string>
          isActive: boolean
        }
        internal: {
          contentFilePath: string
        }
      }[]
    }
  }
  errors?: any
}

export const createPages: GatsbyNode['createPages'] = async ({
  graphql,
  actions,
  reporter,
}) => {
  const { createPage } = actions

  const result: MdxData = await graphql(`
    {
      allMdx(
        # filter: { frontmatter: { isActive: { ne: false } } }
        sort: { frontmatter: { date: DESC } }
      ) {
        nodes {
          id
          fields {
            slug
          }
          frontmatter {
            type
            related
          }
          internal {
            contentFilePath
          }
        }
      }
    }
  `)

  if (result.errors) {
    reporter.panic(`Error loading posts`, result.errors)
    throw result.errors
  }

  if (!result.data) {
    reporter.panic(`No Data`, result.errors)
    throw 'No data found'
  }

  createPage({
    path: `/`,
    component: path.resolve('./src/pages/home.tsx'),
  })

  const postTemplate = path.resolve('./src/templates/post.tsx')
  const createPostPromise = result.data?.allMdx.nodes.map(post => {
    createPage({
      path: `${post.fields.slug}`,
      component: `${postTemplate}?__contentFilePath=${post.internal.contentFilePath}`,
      context: {
        slug: post.fields.slug,
        id: post.id,
        relatedPosts: post.frontmatter.related ? post.frontmatter.related : [],
        // anything else you want to pass to your context
      },
    })
  })

  await Promise.all([createPostPromise])
}
