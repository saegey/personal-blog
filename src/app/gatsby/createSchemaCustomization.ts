import type { GatsbyNode } from 'gatsby'

export const createSchemaCustomization: GatsbyNode['createSchemaCustomization'] =
  ({ actions }) => {
    const { createTypes } = actions

    createTypes(`
    type SiteSiteMetadata {
      author: Author
      siteUrl: String
      social: Social
    }

    type Author {
      name: String
      summary: String
    }

    type Social {
      twitter: String
      github: String
      strava: String
      instagram: String
    }

    type Mdx implements Node {
      frontmatter: Frontmatter!
      fields: Fields
      gpxData: File @link(by: "name", from: "frontmatter.gpxFile")
      results: File @link(by: "name", from: "frontmatter.results.file")
      segments: File @link(by: "name", from: "frontmatter.trainingPeaks")
    }

    type Frontmatter {
      id: String!
      title: String!
      description: String
      date: Date! @dateformat
      publishedDate: Date @dateformat
      tags: [String!]
      location: String
      type: String!
      isActive: Boolean
      related: [String]
      subType: String
    }

    type Coordinate {
      x: String,
      y: String
    }

    type Fields {
      slug: String
    }

  `)
  }
