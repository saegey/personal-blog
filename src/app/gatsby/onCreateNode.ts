import type { GatsbyNode } from 'gatsby'
import { createFilePath } from 'gatsby-source-filesystem'
import { slugify } from '../../lib/util'
import { handleJsonNode } from './parsers/jsonParser'
import { handleTextNode } from './parsers/textParser'
import { handleGpxNode } from './parsers/gpxParser'

type NodeType = {
  internal: {
    type: string
    description: string
    mediaType?: string
  }
  frontmatter: {
    type: string
  }
}

export const onCreateNode: GatsbyNode['onCreateNode'] = async ({
  node,
  actions,
  getNode,
  loadNodeContent,
}) => {
  const { createNodeField } = actions
  const nodeValue = node as any as NodeType

  if (nodeValue.internal.type === 'Mdx') {
    const slug: string = createFilePath({ node, getNode })

    createNodeField({
      name: `slug`,
      node,
      value: `${slugify(nodeValue.frontmatter.type)}${slug}`,
    })
  }

  if (node.internal.mediaType === 'application/json') {
    await handleJsonNode({ node, createNodeField, loadNodeContent });
  }

  if (node.internal.mediaType === 'text/plain') {
    await handleTextNode({ node, createNodeField, loadNodeContent });
  }

  if (
    node.internal.mediaType === 'application/octet-stream' ||
    node.internal.mediaType === 'application/gpx+xml'
  ) {
    await handleGpxNode({ node, createNodeField, loadNodeContent });
  }
}
