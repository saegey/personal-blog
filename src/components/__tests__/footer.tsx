/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'
// import { render } from '@testing-library/react'
import { ThemeProvider } from 'theme-ui'
import * as Gatsby from 'gatsby'

import Footer from '../footer'

import { theme } from './__test-utils__'

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`)
const mockUseStaticQuery = {
  site: {
    siteMetadata: {
      title: `Gatsby Default Starter`,
      social: {
        instagram: 'instagramusername',
        twitter: 'twitterusername',
        github: 'githubusername',
        strava: 'stravausername'
      },
    },
  },
}

describe('Footer', () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery)
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('renders', () => {
    const json = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Footer />
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
