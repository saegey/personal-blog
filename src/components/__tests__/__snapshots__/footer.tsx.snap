// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: bodyfont;
  padding: 20px;
  background-color: muted;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: grid;
  margin-bottom: 10px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
  font-weight: 700;
  font-size: 16px;
  margin-bottom: 10px;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 2.5px;
  font-weight: 300;
  padding-right: 16px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 2.5px;
  padding-right: 16px;
  font-weight: 300;
}

.emotion-7 {
  margin-left: auto;
  margin-right: auto;
}

.emotion-8 {
  padding: 0;
  font-family: bodyfont;
  font-weight: 300;
  bottom: 0;
  margin-top: auto;
  margin-bottom: 10px;
}

<footer
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      Gatsby Default Starter
    </div>
    <div
      className="emotion-3"
    >
      <a
        className="emotion-4"
        href="http://instagram.com/instagramusername"
      >
        Instagram
      </a>
      <a
        className="emotion-5"
        href="http://github.com/githubusername"
      >
        Github
      </a>
      <a
        className="emotion-5"
        href="http://twitter.com/twitterusername"
      >
        Twitter
      </a>
    </div>
  </div>
  <div
    className="emotion-7"
  />
  <div
    className="emotion-8"
  >
    Â© 2023
  </div>
</footer>
`;
